### PR TITLE
[frieren] Weight-decay + dropout regularisation sweep on 5L Fourier PE base

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

A systematic val/test gap (~2.5×) on `vol_p` has been observed across all bengio experiments (e.g., val=4.17% → test~8-12% for SW=2.0 experiments). This gap indicates the model is overfitting on the training distribution and generalising poorly to unseen geometries, particularly in the volume pressure channel.

The current AdamW config uses `weight_decay=1e-4` (very light L2 regularisation) and `model_dropout=0.0` (no dropout). Both are proven levers for closing val/test generalisation gaps in Transformer models. Neither has been touched in any bengio experiment so far.

**Specific mechanism:** `weight_decay` directly penalises large weights that specialise to training geometries. `model_dropout` forces distributed representations, preventing any attention head from memorising geometry-specific cues. A combination of modest weight-decay increase + low dropout (0.05–0.1) may close the vol_p val/test gap without hurting val metrics significantly.

**Why this matters:** If vol_p's val/test gap can be halved (e.g., val=4.5% → test=6.5% rather than 12.9%), this would unlock vol_p as a competitive metric, closing overall abupt to within range of the 4.51% AB-UPT target.

## Instructions

Run three trials sequentially on the **alphonse Wave 1 Fourier PE base** (the same recipe used by `vu4jsiic` / PR #174 — 5L/256d + Fourier PE + T_max=50 + EMA off). We vary `--weight-decay` and `--model-dropout` together, sampling three points on the regularisation surface:

```bash
# Trial A: Increased weight decay only (no dropout)
# wd=1e-3 (10× baseline), dropout=0.0 (unchanged)
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 --epochs 50 \
  --weight-decay 1e-3 \
  --model-dropout 0.0 \
  --no-compile-model \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave6-regularization \
  --wandb-name frieren-reg-A-wd1e3-dp0

# Trial B: Low dropout + modest weight decay
# wd=1e-3, dropout=0.05
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 --epochs 50 \
  --weight-decay 1e-3 \
  --model-dropout 0.05 \
  --no-compile-model \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave6-regularization \
  --wandb-name frieren-reg-B-wd1e3-dp005

# Trial C: Higher dropout (only if A or B show promising trajectory at ep10)
# wd=1e-3, dropout=0.1
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 --epochs 50 \
  --weight-decay 1e-3 \
  --model-dropout 0.1 \
  --no-compile-model \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave6-regularization \
  --wandb-name frieren-reg-C-wd1e3-dp01
```

Run sequentially. **Gate for Trial C:** only run if Trial B ep10 abupt < 9.5%. If Trial A shows abupt regression >0.5pp over the control at ep10, stop and post findings — no need for further trials.

**Important notes:**
- Use 5L/256d (NOT 4L/256d) — we are testing regularisation on the **current leading recipe**, not the old baseline, since we want to confirm whether it closes the val/test gap at the same model size where the gap was observed.
- Compare val metrics carefully: `vol_p` is the primary signal, but watch `abupt` doesn't regress. If wd=1e-3 causes `abupt` to worsen by >1pp at ep10 vs control (alphonse `vu4jsiic` ep10 ~8.5%), it is too aggressive.
- Gate thresholds: ep1 < 20%; ep5 < 13%; ep10 < 11% (these are the running gates used by alphonse `vu4jsiic`).

**What to report:**
1. W&B run IDs and links for each trial
2. Full epoch trajectory table (ep1, 5, 10, 15, 20, 25, 30) showing: abupt, surf_p, vol_p, wsx, wsy, wsz
3. At ep30 best checkpoint: run `python train.py --test-primary-only` (if available) or note that test_primary eval is pending
4. Compare Trial A/B/C trajectories to alphonse `vu4jsiic` at matched epochs to isolate regularisation effect
5. Comment on whether `vol_p` val trajectory improved (even slightly) under regularisation

## Baseline

Current best on bengio (`/target/BASELINE.md`):
- W&B run: `m9775k1v` (alphonse PR #74), ep30, 4L/256d, ContinuousSincosEmbed, weight_decay=1e-4, dropout=0.0
- `val_primary/abupt_axis_mean_rel_l2_pct` = **7.2091%**

Reference running experiment (same base, no regularisation):
- W&B run: `vu4jsiic` (alphonse PR #174), 5L/256d + Fourier PE + T_max=50 + EMA off
- ep16 metrics: abupt=7.667%, surf=5.04%, vol=4.39%, wsy=9.95%, wsz=11.50%
- **This is the recipe you are regularising — compare your ep10/ep16 abupt to these numbers**

Key metrics to beat:
- `val_primary/abupt_axis_mean_rel_l2_pct` < **7.2091%** (current best val)
- `val_primary/volume_pressure_rel_l2_pct`: currently ~4.4% val on the running 5L recipe — we want this to **stay low on test** (val/test gap closure is the signal)
- `val_primary/wall_shear_y_rel_l2_pct` target: < 9.10% (binding constraint)
- `val_primary/wall_shear_z_rel_l2_pct` target: < 10.87% (binding constraint)

```bash
# Reproduce baseline
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 30 \
  --no-compile-model \
  --wandb-group bengio-wave2
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT frieren:

Picking up the regularisation sweep. Two notes before launching Trial A:

1. **Adding `--lr-warmup-epochs 5` to all three trial commands** to match the vu4jsiic recipe exactly. The reference run (`vu4jsiic` config: `lr_warmup_epochs=5`) uses 5 warmup epochs. The trial commands in the PR body omit this flag, which would default to `lr_warmup_epochs=0`. Since the goal is to isolate the regularization effect (wd/dropout) against vu4jsiic, the cleanest comparison is to match warmup as well — particularly important because wd=1e-3 at full LR from step 0 is exactly the regime where warmup matters. I'll note this deviation in the results table.

2. **Trial A launching now.** Reference vu4jsiic (running in parallel): ep18 abupt=7.662%, vol=4.34%, surf=5.02%, wsy=9.93%, wsz=11.54%. Each epoch ~34min on this hardware → ~28h per 50-epoch trial. Will run sequentially as instructed; gating decisions at ep10 per PR body.

Will post results trajectory once Trial A reaches ep10 (~6h).

**Comment by morganmcg1** [comment]
ADVISOR: Good catch on the warmup — yes, add `--lr-warmup-epochs 5` to all three trials to match vu4jsiic exactly. The regularization effect is what we're isolating, so any deviation in training dynamics would confound the comparison. The 5-epoch warmup is part of the base recipe and should be held constant.

One flag to note: the ep10 gate is especially important here. If Trial A (wd=1e-3, no dropout) shows abupt regression >0.5pp vs vu4jsiic at ep10 (~8.5%), stop and report — don't proceed to B or C. We want to know whether wd=1e-3 alone is too aggressive before adding dropout on top.

Good luck with Trial A — looking forward to seeing the vol_p trajectory.

**Comment by morganmcg1** [comment]
ADVISOR: Status check — it's been ~3h15m since I confirmed your warmup decision (00:18Z) and the PR has had no updates since. The original ETA was 'Will post results trajectory once Trial A reaches ep10 (~6h)' from 23:58Z, which would be ~05:58Z, so we're still inside the original ep10 window. But please confirm:

1. **Trial A launched?** Post the W&B run ID so I can pull live metrics.
2. **Config sanity check**: `--lr-warmup-epochs 5` added, wd=1e-3, no dropout, all other flags matching vu4jsiic (4L/256d, no EMA, lr=3e-4, T_max=50, 5L Fourier PE? — re-check the PR body matches your launched recipe).

If Trial A is running, you can ignore this and just post the ep10 update when ready. If not yet launched (e.g. waiting on resource, debugging), let me know what's blocking.

Reminder of the **ep10 stop rule**: if Trial A abupt regression >0.5pp vs vu4jsiic at ep10 (so abupt > ~8.5%), stop and report — don't proceed to Trial B/C. We need to know whether wd=1e-3 alone is too aggressive before stacking dropout.

**Comment by morganmcg1** [comment]
ADVISOR — URGENT: 5h45m of silence since launch decision. Acknowledge within 1h or PR closes.

Timeline:
- 23:58Z May 1: You acknowledged and proposed --lr-warmup-epochs 5 fix.
- 00:18Z May 2: I confirmed warmup fix, asked for run ID.
- 03:39Z May 2: I requested config sanity check — no response.
- **05:30Z May 2 (now): Still no W&B run ID, no ep1 sanity, no acknowledgment.**

Original ep10 ETA was ~05:58Z. We are inside the projected window but with **zero observable progress**. This is now indistinguishable from a non-launched experiment.

## Required within 1 hour (deadline: 06:30Z May 2)

1. Post Trial A W&B run ID.
2. Post ep1 val_abupt (sanity vs vu4jsiic ep1=14.04%). Acceptable range: 13.5–14.5%.
3. Confirm config: lr_warmup_epochs=5, weight_decay=1e-3, dropout=0.0, fourier_pe=True, model_layers=5, model_hidden_dim=256, n_params≈4.5M.

If no acknowledgment by 06:30Z, this PR will be closed and the regularization sweep reassigned to another student. We are 5+h into the experiment window with no observable evidence the experiment is running.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #310 because ADVISOR: Closing due to non-response. This PR has been open since 23:58Z May 1 (over 5.5 hours). You acknowledged the brief and confirmed Trial A launch intent at 23:58Z, but no W&B run ID, no ep1 sanity check, and no acknowledgment of follow-up messages at 03:39Z or the URGENT at 05:19Z. The experiment is indistinguishable from never having been launched.

The regularization sweep (lr_warmup_epochs=5, weight_decay=1e-3) is higher-priority work and needs a student who can execute promptly. Reassigning.

This is not a judgment on the hypothesis — weight-decay regularization on the 5L Fourier base is still worth testing and will be assigned to a fresh student..

**Comment by morganmcg1** [comment]
Codex live diagnosis / retrospective result for the orphaned run.

The PR #310 Trial A run did actually launch, but the student never posted the W&B ID back here before the session got stuck.

Run:
- W&B rank0: `qr95m769` (`frieren-reg-A-wd1e3-dp0-rank0`)
- Group: `bengio-wave6-regularization`
- Config: 5L/256d, `--fourier-pe`, `--lr-cosine-t-max 50`, `--lr-warmup-epochs 5`, `--weight-decay 1e-3`, `--model-dropout 0.0`, no EMA
- State at check: still running from the frieren pod, consuming 4 GPUs

Latest/best validation seen at step `463241` (~epoch 26):

| metric | value |
|---|---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` | 7.9913 |
| `val_primary/surface_pressure_rel_l2_pct` | 5.2670 |
| `val_primary/volume_pressure_rel_l2_pct` | 5.5809 |
| `val_primary/wall_shear_x_rel_l2_pct` | 7.5877 |
| `val_primary/wall_shear_y_rel_l2_pct` | 9.8572 |
| `val_primary/wall_shear_z_rel_l2_pct` | 11.6636 |

Matched-epoch comparison:

| run | ep10 abupt | ep25-ish abupt | best/late read |
|---|---:|---:|---:|
| `qr95m769` Trial A wd=1e-3 | 10.3752 | 8.0187 | 7.9913 at ~ep26 |
| `vu4jsiic` 5L Fourier T50 leader | 8.1280 | 7.4618 | 6.9549 current best/latest |
| `m9775k1v` 4L baseline | 8.2042 | 7.3414 | 7.2091 best |

Read: Trial A clearly failed the PR's ep10 stop rule. The rule said to stop if wd=1e-3 regressed by >0.5pp vs `vu4jsiic` at ep10; it was worse by about +2.25pp. Even after another ~16 epochs it remains ~0.54pp behind `vu4jsiic` at comparable epoch and ~0.78pp behind the current 5L leader's latest value. The component breakdown is also worse across all axes, especially volume pressure.

Recommendation: mark Trial A as a negative result for `weight_decay=1e-3` on the 5L Fourier T50 recipe, do not run Trials B/C from this PR, and kill/restart the orphan process to free frieren. No `test_primary/*` result was produced, so this should not be used for paper-facing claims.
